### DISCUSSION TITLE
Propagate executed command's exit code

### DISCUSSION THREAD

**Discussion by fnurglewitz** [asciinema/discussions#212] [Ideas]
Is it possible to propagate the executed command's ( -c flag) exit code?

**Comment by ku1ik** [comment]
I think it would be possible. However the question is, should we return exit code which says "recording succeeded, regardless of command that was recorded" or "command that was recorded succeeded". I'm not sure what's the good answer here.

**Comment by fnurglewitz** [comment]
@sickill I think it depends on the use case, I would like to record a command output (a functional test which prints a table by re-printing it every time something changes) and then propagate the exit code to let jenkins know if the test passed or not

that's why I was asking

**Comment by ku1ik** [comment]
That's a good use case. We could either propagate that by default or via switch.

**Comment by fnurglewitz** [comment]
probably it would be better to implement a specific switch, this way it's backwards-compatible

**Comment by fnurglewitz** [comment]
@sickill Hi, any chance this feature will be implemented?

If not could you please give us a hint on where/how to modifiy asciinema to implement this feature?

**Comment by ku1ik** [comment]
I don't have ETA for this, sorry. 

As for how to approach this...

The child process is started via execvpe: https://github.com/asciinema/asciinema/blob/develop/asciinema/pty_recorder.py#L114

Not sure how to catch its exit status... The parent process catches SIGCHLD signal: https://github.com/asciinema/asciinema/blob/develop/asciinema/pty_recorder.py#L105

But I don't know if there's any extra info we can get about exiting child process when handling this signal.

**Comment by fnurglewitz** [comment]
Yep, we tried to get it too but got stuck

Usually to get the exit code of a child you use the wait() syscall but I'm not sure if it's possible with python

**Comment by ku1ik** [comment]
I have a draft PR implementing this here https://github.com/asciinema/asciinema/pull/488 . Needs more testing on different OSes though.

**Comment by ku1ik** [comment]
FYI there's new `--return` option in asciinema CLI 3.0 which propagates the exit status.
